### PR TITLE
feat: Email notifications

### PR DESCRIPTION
**Pull request by Todmy**
* add a subscription form to email report
* add the email notification form to the profile page
* show subscription result in modal
* require wallet ownership proof
* add subscribe form to the modal after voting
* change the view of email subscription success modal

### Issues
[project](https://github.com/orgs/snapshot-labs/projects/11/views/10)

### Changes 
This PR includes email notification and subscription management functionality

### How to test
Subscription:

1. Log in to your account
2. Click on your profile in the header navigation
3. In the dropdown, select "SUBSCRIBE"
4. Fill out the email form in the modal
5. Submit
6. Sign the message
7. See success/error message

Subscription management:

1. Log in to your account
2. Click on your profile in the header navigation
3. In the dropdown, select "SUBSCRIBE"
4. Fill out the email form in the modal
5. Submit
6. Sign the message
7. Validate the email with a letter on your email provider
8. Close the form (if you missed the previous step, you need to do it and refresh the page)
9. Click on your profile in the header navigation
10. In the dropdown, select "Manage subscriptions"
11. Update your subscription preferences
12. Sign the message
13. Validate that subscription preferences are changed. It also can be checked in DB. Also, you can ask BE to send a test email.

### To-Do

- [ ] 

### Self-review checklist
- [x] I have performed a full self-review of my changes
- [x] I have tested my changes on a preview deployment
- [x] I have tested my changes on a custom domain
- [x] I have run end-to-end tests `yarn cypress:test:e2e`, and they have passed

### CODE REVIEW

**Comment by zzuziak** [comment]
Hey @Todmy could you possibly provide some Loom previews of the design of all the new features? 🙏

**Comment by Todmy** [comment]
@zzuziak 
Loom https://www.loom.com/share/eb4dfcc5bb2f4d43bcbfc517e6f34ef4

Subscription modal:
Screenshot 2023-06-06 at 21 36 08

Subscription modal success:
Screenshot 2023-06-06 at 21 37 30

Subscription modal fail:
Screenshot 2023-06-06 at 21 36 56

Resend email modal (resend functionality is out of scope):
Screenshot 2023-06-06 at 21 37 46

Manage subscriptions modal:
Screenshot 2023-06-06 at 21 38 28

**Comment by samuveth** [comment]
1. I'm already subscribed with `0xF78108c9BBaF466dd96BE41be728Fe3220b37119` and a wrong email. When I click it I see this: image
2. How do I change my wrongly added email?
3. If the title of the modal is "Resend" there should also be a way to resend.

**Comment by samuveth** [comment]
I tried signing up again with another address and `sam@snapshot.org` and it sent me a confirmation email even though I should already be registered with that email address as I'm getting daily proposal updates to that email.

File: package.json [hunk 1]
```diff
@@ -43,8 +43,8 @@
     "@shutter-network/shutter-crypto": "0.1.0-beta.3",
     "@snapshot-labs/lock": "^0.1.1015",
     "@snapshot-labs/pineapple": "^0.1.0-beta.1",
-    "@snapshot-labs/snapshot.js": "^0.4.93",
-    "@snapshot-labs/tune": "^0.1.25",
+    "@snapshot-labs/snapshot.js": "^0.4.86",
+    "@snapshot-labs/tune": "0.1.24",
```
Comment (samuveth) [inline package.json:47]: Why did you remove `^` and you should use latest version.

File: src/components/ModalEmailResend.vue [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      </div>
+    </template>
+    <div class="m-4 mb-6 text-center">
+      {{ $t('emailResend.description') }}
+
+      <BaseInput
+        v-model="maskedEmail"
+        class="my-4 !pl-[40px]"
+        type="email"
+        :is-disabled="true"
```
Comment (samuveth) [inline src/components/ModalEmailResend.vue:24]:
```suggestion
        is-disabled
```

File: package.json (hunk 1 above)
Comment (samuveth) [reply package.json:47]: The same for snapshot.js, use latest version please

**Comment by wa0x6e** [comment]
> I tried signing up again with another address and sam@snapshot.org and it sent me a confirmation email even though I should already be registered with that email address as I'm getting daily proposal updates to that email.

You can sign up with the same email, as long as the address is different

**Comment by Todmy** [comment]
> 1. I'm already subscribed with `0xF78108c9BBaF466dd96BE41be728Fe3220b37119` and a wrong email. When I click "Subscribe" I see this:
> 
> image
> 2. How do I change my wrongly added email?
> 3. If the title of the modal is "Resend" there should also be a way to resend.

it's currently impossible. This will be moved to the next cycle. It was agreed with @aurelianoB and @wa0x6e. If you want I can temporarily change the title. What would you like me to write there? Maybe `Verify email` or `Check your email`?

**Comment by Todmy** [comment]
> I tried signing up again with another address and `sam@snapshot.org` and it sent me a confirmation email even though I should already be registered with that email address as I'm getting daily proposal updates to that email.

@wa0x6e could you answer this? It is processed on the server and maybe it is expected.

**Comment by samuveth** [comment]
Yes, both alternative title are good for me.

**Comment by samuveth** [comment]
Please fix the conflicts

File: package.json [hunk 3]
```diff
@@ -43,8 +43,8 @@
     "@shutter-network/shutter-crypto": "0.1.0-beta.3",
     "@snapshot-labs/lock": "^0.1.1015",
     "@snapshot-labs/pineapple": "^0.1.0-beta.1",
-    "@snapshot-labs/snapshot.js": "^0.4.93",
-    "@snapshot-labs/tune": "^0.1.25",
+    "@snapshot-labs/snapshot.js": "0.4.94",
+    "@snapshot-labs/tune": "0.1.25",
```
Comment (samuveth) [inline package.json:47]: Still missing the `^`

Comment (ChaituVR) [reply package.json:47]: Looking at current issue I feel like better not to use `^` - hard to revert 🥲

Comment (ChaituVR) [reply package.json:47]: But yea maybe in a separate PR

**Comment by samuveth** [comment]
On manage subscription modal, when I reject the signature the modal should stay open.

File: src/components/ModalEmailResend.vue [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<script setup lang="ts">
```
Comment (samuveth) [inline src/components/ModalEmailResend.vue:1]: Is there still use for this component?

File: src/components/ModalEmailManagement.vue [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+const emit = defineEmits(['close']);
+
+const {
+  loading,
+  clientSubscriptions,
+  updateSubscriptions: update
+} = useEmailSubscription();
+const { t } = useI18n();
+
+const updateSubscriptions = (key, value) => {
```
Comment (samuveth) [inline src/components/ModalEmailManagement.vue:15]: Can you use a different name instead of destructuring the composable import?

File: src/components/ModalPostVote.vue [hunk 6]
```diff
@@ -80,6 +85,14 @@ function share(shareTo: 'twitter' | 'lenster') {
           {{ $t('shareOnLenster') }}
         </BaseButton>
 
+        <BaseButton
```
Comment (samuveth) [inline src/components/ModalPostVote.vue:88]: I'm already subscribed and I still see the "Subscribe" button after voting 
image

File: src/composables/useEmailFetchClient.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  const fetchSubscriptionsDetails = body => {
+    return useEmailFetch('/subscriber').post(body).json();
+  };
+
+  const subscribeWithEmail = async unsignedParams => {
+    let signature;
+    try {
+      signature = await plainSign(unsignedParams, SubscribeSchema);
```
Comment (samuveth) [inline src/composables/useEmailFetchClient.ts:45]: `SubscribeSchema` seems to be undefined here?

**Comment by samuveth** [comment]
When I reject signature when subscribing with my email, we should show the same as when I reject on subscription management. Currently it will show the message in the screenshot below, instead we should show notify error and keep the email field visible.

image

**Review by samuveth — APPROVED** [review]
tACK

File: src/components/ModalEmailSubscription.vue [hunk 8]
```diff
@@ -0,0 +1,99 @@
+<script setup lang="ts">
+defineProps<{
+  open: boolean;
+}>();
+
+type ModalView = 'SUBSCRIBE' | 'SUCCESS';
+
+const emit = defineEmits(['close']);
+const { web3Account } = useWeb3();
+const email = ref('');
```
Comment (samuveth) [inline src/components/ModalEmailSubscription.vue:10]: We should keep refs and composables separate

File: src/components/ModalEmailSubscription.vue [hunk 9]
```diff
@@ -0,0 +1,99 @@
+<script setup lang="ts">
+defineProps<{
+  open: boolean;
+}>();
+
+type ModalView = 'SUBSCRIBE' | 'SUCCESS';
+
+const emit = defineEmits(['close']);
```
Comment (samuveth) [inline src/components/ModalEmailSubscription.vue:8]: emit should come after props

File: src/composables/useEmailSubscription.ts [hunk 10]
```diff
@@ -0,0 +1,93 @@
+import { createSharedComposable } from '@vueuse/core';
+
+const subscriptionTypes = ['summary', 'newProposal', 'closedProposal'] as const;
+type SubscriptionType = (typeof subscriptionTypes)[number];
+type SubscriptionStatus = 'NOT_SUBSCRIBED' | 'VERIFIED' | 'UNVERIFIED';
+
+function useEmailSubscriptionComposable() {
+  const userState = ref<SubscriptionStatus>('NOT_SUBSCRIBED');
+  const error = ref('');
```
Comment (samuveth) [inline src/composables/useEmailSubscription.ts:9]: Same here

File: src/components/ModalEmailSubscription.vue (hunk 8 above)
Comment (wa0x6e) [reply src/components/ModalEmailSubscription.vue:11]: what do you mean ?

**Review by samuveth — APPROVED** [review]
tACK
